test_runner/regress/test_tenants.py

```diff
@@ -414,3 +414,39 @@ def create_bg(delay_ms):
 
     # The tenant should end up active
     wait_until_tenant_active(env.pageserver.http_client(), tenant_id, iterations=10, period=1)
+
+
+def test_pageserver_metrics_many_relations(neon_env_builder: NeonEnvBuilder):
+    """Test for the directory_entries_count metric"""
+
+    neon_env_builder.enable_pageserver_remote_storage(RemoteStorageKind.MOCK_S3)
+
+    env = neon_env_builder.init_start()
+
+    endpoint_tenant = env.endpoints.create_start("main", tenant_id=env.initial_tenant)
+
+    # Not sure why but this many tables creates more relations than our limit
+    TABLE_COUNT = 1600
+    COUNT_AT_LEAST_EXPECTED = 5500
+
+    with closing(endpoint_tenant.connect()) as conn:
+        with conn.cursor() as cur:
+            cur.execute("CREATE TABLE template_tbl(key int primary key, value text);")
+            for i in range(TABLE_COUNT):
+                cur.execute(f"CREATE TABLE tbl_{i}(like template_tbl INCLUDING ALL);")
+    endpoint_tenant.stop()
+
```

Comment:
should there be flush waiting around here with an lsn we got *before* we shutdown the endpoint?

Reply:
Added a `wait_for_wal_insert_lsn`

Reply:
Why that?

Reply:
Discussed -- it would make sense to use `wait_for_wal_insert_lsn` only when reading that lsn from within a transaction then committing, then waiting, otherwise with autocommit we could wait for the next periodic transaction status walrecord, which might take some time. That is at least my best understanding of the situation.